### PR TITLE
[pack] Updated ruleset , cleared up errors, warnings

### PR DESCRIPTION
**Pull request by yojagad**

### CODE REVIEW

File: src/WebJobs.Extensions.RabbitMQ.ruleset [hunk 1]
```diff
@@ -4,72 +4,10 @@
...
-    <Rule Id="CA2236" Action="Warning" />
-    <Rule Id="CA2237" Action="Warning" />
-    <Rule Id="CA2238" Action="Warning" />
-    <Rule Id="CA2240" Action="Warning" />
-    <Rule Id="CA2241" Action="Warning" />
-    <Rule Id="CA2242" Action="Warning" />
-  </Rules>
   <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers">
+	<Rule Id="SA0001" Action="None" />
     <Rule Id="SA1309" Action="None" />
```
Comment (pragnagopa) [inline src/WebJobs.Extensions.RabbitMQ.ruleset:9]: nit:format

File: test/WebJobs.Extensions.RabbitMQ.Samples/WebJobs.Extensions.RabbitMQ.Samples.ruleset [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<RuleSet Name="Microsoft Managed Recommended Rules" Description="These rules focus on the most critical problems in your code, including potential security holes, application crashes, and other important logic and design errors. It is recommended to include this rule set in any custom rule set you create for your projects." ToolsVersion="10.0">
+  <Localization ResourceAssembly="Microsoft.VisualStudio.CodeAnalysis.RuleSets.Strings.dll" ResourceBaseName="Microsoft.VisualStudio.CodeAnalysis.RuleSets.Strings.Localized">
+    <Name Resource="MinimumRecommendedRules_Name" />
+    <Description Resource="MinimumRecommendedRules_Description" />
+  </Localization>
+  <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers">
+	<Rule Id="SA0001" Action="None" />
+    <Rule Id="SA1309" Action="None" />
```
Comment (pragnagopa) [inline test/WebJobs.Extensions.RabbitMQ.Samples/WebJobs.Extensions.RabbitMQ.Samples.ruleset:9]: nit: format

File: test/WebJobs.Extensions.RabbitMQ.Tests/RabbitMQServiceTests.cs [hunk 3]
```diff
@@ -13,7 +13,7 @@ public class RabbitMQServiceTests
     {
         [Theory]
         [InlineData("", "localhost", "guest", "guest", 5672, "", "localhost", "guest", "guest", 5672)]
-        [InlineData("amqp://testUserName:testPassword@11.111.111.11:5672", null, null, null, null, "amqp://testUserName:testPassword@11.111.111.11:5672", "11.111.111.11", "testUserName", "testPassword", 5672)]
+        [InlineData("amqp://testUserName:testPassword@11.111.111.11:5672", null, null, null, 8080, "amqp://testUserName:testPassword@11.111.111.11:5672", "11.111.111.11", "testUserName", "testPassword", 5672)]
```
Comment (pragnagopa) [inline test/WebJobs.Extensions.RabbitMQ.Tests/RabbitMQServiceTests.cs]: was this meant to test null port?

**Review by pragnagopa — COMMENTED** [review]
few nits.

File: test/WebJobs.Extensions.RabbitMQ.Tests/RabbitMQServiceTests.cs (hunk 3 above)
Comment (yojagad) [reply test/WebJobs.Extensions.RabbitMQ.Tests/RabbitMQServiceTests.cs]: This stood out to me too. If this was indeed used to test null port, shouldn't GetConnectionFactory have taken a nullable int ?

Comment (pragnagopa) [reply test/WebJobs.Extensions.RabbitMQ.Tests/RabbitMQServiceTests.cs]: I think we default to a known port it not supplied. Can you check?

Comment (yojagad) [reply test/WebJobs.Extensions.RabbitMQ.Tests/RabbitMQServiceTests.cs]: Yeah, it comes from https://github.com/Azure/azure-functions-rabbitmq-extension/blob/dev/src/Bindings/RabbitMQClientBuilder.cs#L32. Looks like it defaults to 0, so I updated test for the default value.

File: test/WebJobs.Extensions.RabbitMQ.Tests/RabbitMQServiceTests.cs [hunk 4]
```diff
@@ -13,7 +13,7 @@ public class RabbitMQServiceTests
     {
         [Theory]
         [InlineData("", "localhost", "guest", "guest", 5672, "", "localhost", "guest", "guest", 5672)]
-        [InlineData("amqp://testUserName:testPassword@11.111.111.11:5672", null, null, null, 8080, "amqp://testUserName:testPassword@11.111.111.11:5672", "11.111.111.11", "testUserName", "testPassword", 5672)]
+        [InlineData("amqp://testUserName:testPassword@11.111.111.11:5672", null, null, null, 0, "amqp://testUserName:testPassword@11.111.111.11:5672", "11.111.111.11", "testUserName", "testPassword", 5672)]
```
Comment (pragnagopa) [inline test/WebJobs.Extensions.RabbitMQ.Tests/RabbitMQServiceTests.cs]: Leave it at null? i think code needs to take care of picking default port

Comment (yojagad) [reply test/WebJobs.Extensions.RabbitMQ.Tests/RabbitMQServiceTests.cs]: reverted it to null. Added an additional ruleset too.

**Review by pragnagopa — APPROVED** [review]

**Pull request merged**